prometheus-connector/README.md

```diff
@@ -0,0 +1,189 @@
+
+1. Input time-series must be well-formed, e.g. only valid and non-duplicated labels, 
+samples in timestamp order (see [Labels and Ordering](https://prometheus.io/docs/concepts/remote_write_spec/#labels) in Prometheus Remote-Write specs).
+2. Input time-series with identical labels are sent to the sink in timestamp order.
+3. If sink parallelism > 1 is used, the input stream must be partitioned so that all time-series with identical labels go to the same sink subtask. A `KeySelector` is provided to partition input correctly (see [Partitioning](#partitioning), below). 
+
+
+#### Sink input objects
+
+To help sending well-formed data to the sink, the connector expect [`PrometheusTimeSeries`](./src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusTimeSeries.java) POJOs as input.
+
+Each `PrometheusTimeSeries` instance maps 1-to-1 to a [remote-write `TimeSeries`](https://prometheus.io/docs/concepts/remote_write_spec/#protocol). Each object contains:
+* exactly one `metericName`, mapped to the special  `__name__` label
+* optionally, any number of additional labels { k: String, v:String }
+* one or more `Samples` { value: double, timestamp: long } - must be in timestamp order
+
+`PrometheusTimeSeries` provides a builder interface.
+
+```java
+
+// List<Tuple2<Double, Long>> samples = ...
+
+PrometheusTimeSeries.Builder tsBuilder = PrometheusTimeSeries.builder()
+    .withMetricName("CPU") // mapped to  `__name__` label
+    .addLabel("InstanceID", instanceId)
+    .addLabel("AcccountID", accountId);
+    
+for(Tuple2<Double, Long> sample : samples) {
+    tsBuilder.addSample(sample.f0, sample.f1);
+}
+
+PrometheusTimeSeries ts = tsBuilder.build();
+```
+
+
+**Important**: for efficiency, the builder does reorder the samples. It is responsibility of the application to **add samples in timestamp order**.
+
+### Batching, blocking writes and retry
+
+The sink batches multiple time-series into a single write-request, retaining the order..
+
+Batching is based on the number of samples. Each write-request contains up to 500 samples, with a max buffering time of 5 seconds 
+(both configurable). The number of time-series doesn't matter.
+
+As by [Prometheus Remote-Write specifications](https://prometheus.io/docs/concepts/remote_write_spec/#retries-backoff), 
+the sink retries 5xx and 429 responses. Retrying is blocking, to retain sample ordering, and uses and exponential backoff.
+
+The exponential backoff starts with an initial delay (default 30 ms) and increases it exponentially up to a max retry 
+delay (default 5 sec). It continues retrying until the max number of retries is reached (default reties forever).
+
+On non-retriable error response (4xx, except 429, non retryable exceptions), or on reaching the retry limit, 
+**the entire write-request**, containing the batch of time-series, **is dropped**.
+
+Every dropped request is logged at WARN level, including the reason provided by the remote-write endpoint.
```

Comment:
We can mention the configuration here too

Reply:
Added explanation